usd_qtpy/render_util/dialog.py

```diff
@@ -0,0 +1,889 @@
+        self.vlayout.addWidget(self.progressbar)
+
+    def _update_fit(self):
+        if "New: Framing Camera" in self.cbox_camera.currentText():
+            self.spinbox_fit.setDisabled(False)
+            self.lbl_fit.setDisabled(False)
+        else:
+            self.spinbox_fit.setDisabled(True)
+            self.lbl_fit.setDisabled(True)
+
+    def _update_resolution(self, res_tuple: tuple[int,int]):
+        self.spinbox_horresolution.setValue(res_tuple[0])
+        self.spinbox_verresolution.setValue(res_tuple[1])
+
+    def _update_framerange_options(self):
+        index = self.cbox_framerange_options.currentIndex()
+        if index == 0:
+            self.spinbox_frame_start.setDisabled(False)
+            self.spinbox_frame_end.setDisabled(True)
+            self.spinbox_frame_stride.setDisabled(True)
+        elif index == 1:
+            self.spinbox_frame_start.setDisabled(False)
+            self.spinbox_frame_end.setDisabled(False)
+            self.spinbox_frame_stride.setDisabled(False)
+        elif index == 2:
+            self.spinbox_frame_start.setDisabled(True)
+            self.spinbox_frame_end.setDisabled(True)
+            self.spinbox_frame_stride.setDisabled(True)
+
+    def _prompt_renderoutput(self):
+        filename = prompt_output_path("Render result to...")
+        if filename:
+            self.txt_filename.setText(os.path.normpath(filename))
+
+    def _gather_purposes(self) -> list[str]:
+        purposes = []
+        
+        if self.chk_purpose_default:
+            purposes.append("default")
+        
+        if self.chk_purpose_render:
+            purposes.append("render")
+
+        if self.chk_purpose_proxy:
+            purposes.append("proxy")
+
+        if self.chk_purpose_guide:
+            purposes.append("guide")
+        
+        return purposes
+
+    def _set_total_frames(self, frames: int):
+        """
+        Update progress bar with reported maximum frames
+        """
+        #self._total_frames = frames
+        self.progressbar.setFormat(f"Rendering %v / {frames} frames...")
+        self.progressbar.setMinimum(1)
+        self.progressbar.setMaximum(frames)
+    
+    def _set_render_progress(self, frame: int):
+        """
+        Increment progress bar with reported rendered frames.
+        """
+        #self._render_progress = frame
+        self.progressbar.setValue(frame)
+
+    def _construct_frames_argument(self) -> str:
+        start = self.spinbox_frame_start.value()
+        end = self.spinbox_frame_end.value()
+        stride = self.spinbox_frame_stride.value()
+
+        opt_index = self.cbox_framerange_options.currentIndex()
+
+        if opt_index == 0:
+            return f"{start}"
+        elif opt_index == 1:
+            return playblast.get_frames_string(start,end,stride)  
+        elif opt_index == 2:
+            if self._stageview:
+                return playblast.get_stageview_frame(self._stageview)      
```

Comment:
This could be clearer if commented or checking against constants?

Reply:
I don't like setting up constants in classes, especially not for this particular element since this element doesn't
stay the same when inherited, so I'll comment on it. 

Normally I'd define an enum, if something is common.